test/steps/run_steps.go

```diff
@@ -16,9 +15,15 @@ func RunSteps(suite *godog.Suite, fs *FeatureState) {
 		return nil
 	})
 
-	suite.Step(`^it runs the commands$`, func(table *gherkin.DataTable) error {
+	suite.Step(`^it runs the commands$`, func(input *gherkin.DataTable) error {
 		commands := test.GitCommandsInGitTownOutput(fs.activeScenarioState.lastRunOutput)
-		return gherkintools.EnsureStringSliceMatchesTable(commands, table)
+		table := test.RenderExecutedGitCommands(commands, input)
+		diff, errorCount := table.Equal(input)
+		if errorCount != 0 {
+			fmt.Println(diff)
+			return fmt.Errorf("found %d differences", errorCount)
```

Comment:
Is there any difference between printing the diff separately and having the diff as part of the error message?

Reply:
Only a more visual separation since the diff is usually large. Based on your suggestion I am now printing the diffs inline with the other error message, separated by a manual newline in the error message. Looks cleaner in the code!